### PR TITLE
Add nested filters to endpoints creation

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

File: app/controllers/endpoints_controller.rb [hunk 1]
```diff
@@ -21,6 +21,11 @@ def create
     endpoint = Endpoint.new(endpoint_params)
     endpoint.account = current_user.account
     endpoint.type ||= Endpoint.name
+
+    nested_filters.each do |filter_params|
```
Comment (adamruzicka) [inline app/controllers/endpoints_controller.rb]: `filter_params` is a two element array (like `[:app_ids, [1, 2, 3]]`).

Changing this to:
```ruby
authorize endpoint.filters.build(nested_filters.merge(account: current_user.account))
```
seems to work for `app_ids` and `event_type_ids`, but fails with `ActiveRecord::AssociationTypeMismatch: SeverityFilter(#70042269683180) expected, got "low" which is an instance of String(#47374285323740)` for severity filters

**Comment by ShimShtein** [comment]
@adamruzicka, finally ready for review. Beefed up endpoint tests to handle nested filters.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -35,7 +36,11 @@ def render_update(record, serializer_class)
     if yield record
       render :json => serializer_class.new(record)
     else
-      render :json => { :errors => record.errors }, :status => :unprocessable_entity
+      render_unprocessible_entity record.errors
     end
   end
+
+  def render_unprocessible_entity(errors)
```
Comment (adamruzicka) [inline app/controllers/application_controller.rb]: `s/ible/able/` just to keep things consistent

**Comment by ShimShtein** [comment]
@adamruzicka Ready!

**Comment by adamruzicka** [comment]
Thanks @ShimShtein !

**Pull request merged**